src/olympia/reviewers/utils.py

```diff
@@ -486,16 +487,13 @@ def get_actions(self):
             and self.version.human_review_date
         )
         addon_is_valid = self.addon.is_public() or self.addon.is_unreviewed()
-        addon_is_valid_and_version_is_listed = (
-            addon_is_valid
-            and self.version
-            and self.version.channel == amo.CHANNEL_LISTED
-        )
-        current_version_is_listed_and_auto_approved = (
-            self.version
-            and self.version.channel == amo.CHANNEL_LISTED
-            and self.addon.current_version
-            and self.addon.current_version.was_auto_approved
+        addon_is_valid_and_version_is_listed = addon_is_valid and version_is_listed
+        current_or_latest_listed_version_was_auto_approved = version_is_listed and (
+            (
+                self.addon.current_version
+                and self.addon.current_version.was_auto_approved
+            )
+            or self.version.was_auto_approved
```

Comment:
This sets up an inconsistency in an add-on that has a current_version that wasn't auto_approved but `self.version` _was_ auto approved, because later in `confirm_auto_approved` we only drop back to `self.version` if `current_version` is falsey.  I'm not sure if this case can happen, but perhaps it should be changed to something similar to:
```suggestion
        current_or_latest_listed_version_was_auto_approved = version_is_listed and (
            (
                self.addon.current_version
                and self.addon.current_version.was_auto_approved
            )
            or (not self.addon.current_version and self.version.was_auto_approved)
```

Reply:
Yeah. But if `current_version` is not falsey, then that is the version we should care about. If looking at the listed channel, it's only really absent when the add-on is deleted.

Reply:
Ah yeah I see. I'll tweak the condition.